Sources/MongoSwift/MongoClient.swift

```diff
@@ -402,6 +461,105 @@ public class MongoClient {
         )
     }
 
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+    @available(macOS 10.15, *)
+    private class CmdHandler: CommandEventHandler {
+        private var con: AsyncStream<CommandEvent>.Continuation?
+        internal init(con: AsyncStream<CommandEvent>.Continuation) {
+            self.con = con
+        }
+
+        internal init() {
+            self.con = nil
+        }
+
+        // Satisfies the protocol
+        internal func handleCommandEvent(_ event: CommandEvent) {
+            self.con?.yield(event)
+        }
+
+        internal func setCon(con: AsyncStream<CommandEvent>.Continuation) {
+            self.con = con
+        }
+    }
+
+    @available(macOS 10.15, *)
+    private class SDAMHandler: SDAMEventHandler {
+        private var con: AsyncStream<SDAMEvent>.Continuation?
+        internal init(con: AsyncStream<SDAMEvent>.Continuation) {
+            self.con = con
+        }
+
+        internal init() {
+            self.con = nil
+        }
+
+        // Satisfies the protocol
+        internal func handleSDAMEvent(_ event: SDAMEvent) {
+            self.con?.yield(event)
+        }
+
+        internal func setCon(con: AsyncStream<SDAMEvent>.Continuation) {
+            self.con = con
+        }
+    }
+
+    /// Provides an `AsyncSequence` API for consuming command monitoring events.
+    /// Example: printing the command events out would be written as
+    /// c
+    /// Wrapping in a `Task { ... }` may be desired for asynchronicity.
+    /// Note that only the most recent 100 events are stored in the stream.
+    @available(macOS 10.15, *)
+    public func commandEventStream() -> CommandEventStream {
+        let handler = CmdHandler()
+        var outerCon: AsyncStream<CommandEvent>.Continuation?
+        var commandEvents = CommandEventStream(
+            stream: AsyncStream(
+                CommandEvent.self,
+                bufferingPolicy: .bufferingNewest(100)
+            ) { con in
+                handler.setCon(con: con)
+                self.addCommandEventHandler(handler)
+                outerCon = con
+                if self.wasClosed {
```

Comment:
Agree that the method inside only works when called on a closed client which doesnt seem problematic. Especially for now, I think that we should just ignore a call on a closed client as u prescribe. 

Agree with implementation of looping over handlers in finish. I changed the type of `con` to internal and made the `finish()` method on the handler. I also just renamed it to `continuation` for the symmetry with the code in `APM.swift`. Wrote 2 loops in `MongoClient.close()` to finish the corresponding handlers. I noticed the `syncClose()` method, is that relevant to the issue at hand? I imagine its not (especially bc its a sync method).

Will look into writing such tests. I noticed that manually closing the client correctly proved to be problematic given the current test client but maybe I can rewrite.

Reply:
ah good question, `syncClose()` is kind of a funny situation - it is a synchronous method but it is part of the async API. it exists because (despite writing async applications) users may often be doing client setup and teardown in synchronous contexts -- for example, you might `defer { client.syncClose() }`, or call `client.syncClose()`  in a `deinit` method, and neither of those contexts directly allow calling an async method. we wanted to provide an easy way for people to do that type of cleanup if they need it. (and we use it a lot ourselves in tests.)

so, it's possible async API users are using `syncClose()` for cleanup, and since it doesn't directly wrap the async `close()` method I think we will want to add the same logic for finishing the continuations in that method too. 

>  I noticed that manually closing the client correctly proved to be problematic given the current test client but maybe I can rewrite.
I think you would not be able to use `withTestClient` in these cases, since that method already closes the client once the closure you provide to it finishes executing. maybe that's what you're running into here? instead, you could manually do `makeTestClient()`, which will allow you to close it at the end.

Reply:
Done. provided cleanup in both `.close()` and `.syncClose()`. You were right in that using `.withTestClient` provided some problems with expected behavior of tasks upon closing, so I used the internal logic of that method to create a client when we were explictly testing the presence of events in tasks. For other tests, using the `.withTestClient` seemed to be effective